modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/StorageSortedIndexDescriptor.java

```diff
@@ -99,34 +95,30 @@ public String toString() {
 
     private final BinaryTupleSchema binaryTupleSchema;
 
+    private final boolean pk;
+
     /**
      * Constructor.
      *
      * @param table Catalog table descriptor.
      * @param index Catalog index descriptor.
      */
     public StorageSortedIndexDescriptor(CatalogTableDescriptor table, CatalogSortedIndexDescriptor index) {
-        this(index.id(), extractIndexColumnsConfiguration(table, index));
+        this(index.id(), extractIndexColumnsConfiguration(table, index), table.primaryKeyIndexId() == index.id());
     }
 
     /**
      * Creates an Index Descriptor from a given set of column descriptors.
      *
      * @param indexId Index ID.
      * @param columnDescriptors Column descriptors.
+     * @param pk Primary index flag.
      */
-    public StorageSortedIndexDescriptor(int indexId, List<StorageSortedIndexColumnDescriptor> columnDescriptors) {
+    public StorageSortedIndexDescriptor(int indexId, List<StorageSortedIndexColumnDescriptor> columnDescriptors, boolean pk) {
```

Comment:
I have the same list of questions:
- if this constructor has to be public for some reason, then please make it `@TestOnly`
- why does it have to be an internal state rather than an external check? This is a calculable property, after all

Reply:
It is also used in product code, please see usages.
Please give an example of how you could implement your idea in those places where I use this internal flag.